app/lib/actions/katello/host/remove_subscriptions.rb

```diff
@@ -14,16 +14,18 @@ def plan(host, pools_with_quantities_params)
             cp_consumer.filter_entitlements(pool_with_quantities.pool.cp_id, pool_with_quantities.quantities)
           end
 
+          pool_ids = []
           entitlements.flatten.each do |entitlement|
+            pool_ids << entitlement['pool']['id']
             plan_action(::Actions::Candlepin::Consumer::RemoveSubscription, :uuid => host.subscription_facet.uuid,
                         :entitlement_id => entitlement['id'], :pool_id => entitlement['pool']['id'])
-            plan_self(:host_name => host.name)
           end
+          plan_self(:host => host, :host_name => host.name, :pool_ids => pool_ids)
         end
 
         def finalize
-          ::Katello::Pool.where(:id => input[:pool_ids]).each(&:import_data)
-          host = ::Host.find_by(:id => input[:managed][:id])
+          ::Katello::Pool.where(:cp_id => input[:pool_ids]).each(&:import_data)
+          host = ::Host.find_by(:id => input[:host][:id])
```

Comment:
It's a timing thing: The results returned to the UI are not up to date before the candlepin event arrives. This PR updates the local katello data immediately.